app/models/base.rb

```diff
@@ -20,13 +20,13 @@ def refresh_database
 
   def service_ids(appliance_id, appdb_data)
     appdb_data.collect do |service|
-      service['id'] if service['appliance'].any? { |appliance| appliance['id'] == appliance_id }
+      service['id'] if service['appliances'].any? { |appliance| appliance['id'] == appliance_id }
     end.compact
   end
 
   def service_ids_old(appliance_id, appdb_data)
     appdb_data.select! do |site|
-      !site['appliance'].select! { |appliance| appliance['id'] == appliance_id }.blank?
+      !site['appliances'].select! { |appliance| appliance['id'] == appliance_id }.blank?
```

Comment:
This line is really difficult to read and understand, especially the `!`...`.blank?` combo miles apart. Also, you should never modify passed arguments like that. I'm talking about `appdb_data` and those two `select!` calls. !resolve